examples/bond/scripts/run/bond-functions.js

```diff
@@ -0,0 +1,328 @@
+
+  const buybackTx = {
+    type: types.TransactionType.CallNoOpSSC,
+    sign: types.SignType.SecretKey,
+    fromAccount: storeManagerAccount,
+    appID: appInfo.appID,
+    payFlags: {},
+    appArgs: ['str:create_buyback', convert.addressToPk(buybackLsig.address())]
+  };
+
+  // Only store manager can allow opt-in to ASA for lsig
+  const optInTx = [
+    {
+      type: types.TransactionType.TransferAlgo,
+      sign: types.SignType.SecretKey,
+      fromAccount: storeManagerAccount,
+      toAccountAddr: buybackLsig.address(),
+      amountMicroAlgos: 0,
+      payFlags: {}
+    },
+    {
+      type: types.TransactionType.OptInASA,
+      sign: types.SignType.LogicSignature,
+      fromAccountAddr: buybackLsig.address(),
+      lsig: buybackLsig,
+      assetID: newAsaInfo[assetID],
+      payFlags: {}
+    }
+  ];
+  await executeTransaction(deployer, optInTx);
+
+  console.log('Storing buyback address!');
+  await executeTransaction(deployer, buybackTx);
+  console.log('Buyback address stored successfully!');
+}
+
+// Buyer's exit from bond
+async function exitBuyer (deployer, buyerAccount) {
+  const exitAmount = 10 * 1000 - 1000;
+  const exitTx = [
+    //  Bond token transfer to buyback address
+    {
+      type: types.TransactionType.TransferAsset,
+      sign: types.SignType.SecretKey,
+      fromAccount: buyerAccount,
+      toAccountAddr: buybackLsig.address(),
+      amount: 10,
+      assetID: newAsaInfo[assetID],
+      payFlags: { totalFee: 1000 }
+    },
+    // Nominal price * amount paid to buyer
+    {
+      type: types.TransactionType.TransferAlgo,
+      sign: types.SignType.LogicSignature,
+      fromAccountAddr: buybackLsig.address(),
+      lsig: buybackLsig,
+      toAccountAddr: buyerAccount.addr,
+      amountMicroAlgos: exitAmount,
+      payFlags: { totalFee: 1000 }
+    },
+    // call to bond-dapp
+    {
+      type: types.TransactionType.CallNoOpSSC,
+      sign: types.SignType.SecretKey,
+      fromAccount: buyerAccount,
+      appID: appInfo.appID,
+      payFlags: {},
+      appArgs: ['str:exit']
+    }
+  ];
+
+  console.log('Exiting');
+  await executeTransaction(deployer, exitTx);
+  console.log('Exited');
+}
+
+async function run (runtimeEnv, deployer) {
+  const masterAccount = deployer.accountsByName.get('master-account');
+  const creatorAccount = deployer.accountsByName.get('john');
+  const storeManagerAccount = deployer.accountsByName.get('alice');
+  const buyerAccount = deployer.accountsByName.get('bob');
+
+  // Create DEX, burn B_0, issue B_1
+  await createDex(deployer, masterAccount, creatorAccount, storeManagerAccount);
+
+  // Redeem coupon_value
+  await redeem(deployer, buyerAccount, storeManagerAccount);
+
+  // create buyback
+  await createBuyback(deployer, storeManagerAccount);
+
+  // exit buyer from bond, buyer can exit only if maturity period is over
+  // currently set to 1000 seconds
+  await exitBuyer(deployer, buyerAccount);
```

Comment:
let's move `exit` and `createBuyback` to another `run` script (both can be in the same script).

Reply:
ok